### PR TITLE
[CoreNFC] Add CoreNFC Xcode 9 Beta 1 Bindings

### PR DESCRIPTION
**Pull request by dalexsoto**
CoreNFC is not exposed in iOS simulator, no framework/headers are found
makes sense since there is no NFC chip on it.

CoreNFC says it is available on tvOS but headers says just 3 enums
are so not enabling tvOS in this first beta.

### CODE REVIEW

**Comment by dalexsoto** [comment]
Mmm looks like I missed a check somewhere

File: tests/introspection/ApiSignatureTest.cs [hunk 1]
```diff
@@ -669,6 +669,8 @@ protected virtual bool Check (char encodedType, Type type)
 				case "System.Byte":
 				// GLKBaseEffect 'instance Boolean get_ColorMaterialEnabled()' selector: colorMaterialEnabled == C8@0:4
 				case "System.Boolean":
+				// CoreNFC.NFCTypeNameFormat enum is byte
+				case "CoreNFC.NFCTypeNameFormat":
```
Comment (spouliot) [inline tests/introspection/ApiSignatureTest.cs]: What was the error here ?
Afaik this convertion should be handled by the tests, it's not the first byte based enum

Comment (dalexsoto) [inline tests/introspection/ApiSignatureTest.cs]: I tried adding 
```
if (type.IsEnum)
	type = type.GetEnumUnderlyingType ();
```

But that ended in the following

```
Return Value of selector: accessibilityTraits on type UIKit.UIView, Type: UIKit.UIAccessibilityTrait, Encoded as: Q
Signature failure in UIKit.UIView setAccessibilityTraits: Parameter 'value' (#1) is encoded as 'Q' and bound as 'UIKit.UIAccessibilityTrait'
Signature failure in AVFoundation.AVAudioSession setAggregatedIOPreference:error: Parameter 'ioType' (#1) is encoded as 'Q' and bound as 'AVFoundation.AVAudioSessionIOType'
Signature failure in Foundation.NSAttributedString drawWithRect:options:context: Parameter 'options' (#2) is encoded as 'q' and bound as 'Foundation.NSStringDrawingOptions'
Signature failure in Foundation.NSAttributedString boundingRectWithSize:options:context: Parameter 'options' (#2) is encoded as 'q' and bound as 'Foundation.NSStringDrawingOptions'
Return Value of selector: unit on type AudioUnit.AUParameter, Type: AudioUnit.AudioUnitParameterUnit, Encoded as: I
Return Value of selector: headingOrientation on type CoreLocation.CLLocationManager, Type: CoreLocation.CLDeviceOrientation, Encoded as: i
Signature failure in CoreLocation.CLLocationManager setHeadingOrientation: Parameter 'value' (#1) is encoded as 'i' and bound as 'CoreLocation.CLDeviceOrientation'
Return Value of selector: authorizationStatus on type CoreLocation.CLLocationManager, Type: CoreLocation.CLAuthorizationStatus, Encoded as: i
Return Value of selector: accessibilityTraits on type UIKit.UIBarItem, Type: UIKit.UIAccessibilityTrait, Encoded as: Q
Signature failure in UIKit.UIBarItem setAccessibilityTraits: Parameter 'value' (#1) is encoded as 'Q' and bound as 'UIKit.UIAccessibilityTrait'
Return Value of selector: accessibilityTraits on type UIKit.UIImage, Type: UIKit.UIAccessibilityTrait, Encoded as: Q
Signature failure in UIKit.UIImage setAccessibilityTraits: Parameter 'value' (#1) is encoded as 'Q' and bound as 'UIKit.UIAccessibilityTrait'
Signature failure in AudioUnit.AUParameterTree createParameterWithIdentifier:name:address:min:max:unit:unitName:flags:valueStrings:dependentParameters: Parameter 'unit' (#6) is encoded as 'I' and bound as 'AudioUnit.AudioUnitParameterUnit'
Signature failure in MapKit.MKMapView setUserTrackingMode:animated: Parameter 'trackingMode' (#1) is encoded as 'q' and bound as 'MapKit.MKUserTrackingMode'
Return Value of selector: userTrackingMode on type MapKit.MKMapView, Type: MapKit.MKUserTrackingMode, Encoded as: q
Signature failure in MapKit.MKMapView setUserTrackingMode: Parameter 'value' (#1) is encoded as 'q' and bound as 'MapKit.MKUserTrackingMode'
Return Value of selector: target on type GLKit.GLKTextureInfo, Type: GLKit.GLKTextureTarget, Encoded as: I
Return Value of selector: target on type GLKit.GLKEffectPropertyTexture, Type: GLKit.GLKTextureTarget, Encoded as: I
Signature failure in GLKit.GLKEffectPropertyTexture setTarget: Parameter 'value' (#1) is encoded as 'I' and bound as 'GLKit.GLKTextureTarget'
Return Value of selector: orientation on type Photos.PHLivePhotoEditingContext, Type: ImageIO.CGImagePropertyOrientation, Encoded as: I
Return Value of selector: playlistAttributes on type MediaPlayer.MPMediaPlaylist, Type: MediaPlayer.MPMediaPlaylistAttribute, Encoded as: Q
Signature failure in SpriteKit.SKMutableTexture initWithSize:pixelFormat: Parameter 'pixelFormat' (#2) is encoded as 'i' and bound as 'CoreVideo.CVPixelFormatType'
Return Value of selector: accessibilityTraits on type WatchKit.WKInterfaceObject, Type: UIKit.UIAccessibilityTrait, Encoded as: Q
Signature failure in WatchKit.WKInterfaceObject setAccessibilityTraits: Parameter 'value' (#1) is encoded as 'Q' and bound as 'UIKit.UIAccessibilityTrait'
```

This needs more love so I opened this bug https://bugzilla.xamarin.com/show_bug.cgi?id=57573 so we do not forget

**Comment by dalexsoto** [comment]
build

**Comment by rolfbjarne** [comment]
Test failures are not related to this PR.

**Pull request merged**